### DISCUSSION TITLE
CF Pages failing to build.

### DISCUSSION THREAD

**Discussion by monstermuffin** [nunocoracao/blowfish#497] [Q&A] [answered]
**Describe the bug**
Upon building using CF pages the following error is displayed and the build fails.

`Error: "/opt/buildhome/repo/themes/blowfish/i18n/zh-TW.yaml:1:1": failed to load translations: unable to parse translation #0 because invalid plural category part_of_series`

**To Reproduce**
1. Create page on CF-Pages
2. Link repo
3. Atempt to build

**Expected behavior**
The build to succeed. 

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Firefox
 - Version 109.0

**Hugo & Blowfish versions**
* Hugo 0.54.0
* Blowfish v2.26.0

**Additional context**
The entire log from CF is as follows:

```
14:43:29.498 | Submodule 'themes/blowfish' (https://github.com/nunocoracao/blowfish.git) registered for path 'themes/blowfish'
-- | --
14:43:29.499 | Cloning into '/opt/buildhome/clone/themes/blowfish'...
14:43:29.499 | Submodule path 'themes/blowfish': checked out '17557c7d731c3f07fb8cf72ebfc11c0cf21faa9a'
14:43:29.499 |  
14:43:29.570 | Success: Finished cloning repository files
14:43:30.263 | Installing dependencies
14:43:30.276 | Python version set to 2.7
14:43:34.347 | v12.18.0 is already installed.
14:43:35.636 | Now using node v12.18.0 (npm v6.14.4)
14:43:35.890 | Started restoring cached build plugins
14:43:35.918 | Finished restoring cached build plugins
14:43:36.490 | Attempting ruby version 2.7.1, read from environment
14:43:40.391 | Using ruby version 2.7.1
14:43:40.767 | Using PHP version 5.6
14:43:40.974 | 5.2 is already installed.
14:43:41.002 | Using Swift version 5.2
14:43:41.003 | Installing Hugo 0.54.0
14:43:41.842 | Hugo Static Site Generator v0.54.0-B1A82C61A/extended linux/amd64 BuildDate: 2019-02-01T10:04:38Z
14:43:41.875 | Started restoring cached go cache
14:43:41.902 | Finished restoring cached go cache
14:43:42.098 | go version go1.14.4 linux/amd64
14:43:42.117 | go version go1.14.4 linux/amd64
14:43:42.122 | Installing missing commands
14:43:42.122 | Verify run directory
14:43:42.122 | Executing user command: hugo -b $CF_PAGES_URL
14:43:42.161 | Error: "/opt/buildhome/repo/themes/blowfish/i18n/zh-TW.yaml:1:1": failed to load translations: unable to parse translation #0 because invalid plural category part_of_series
14:43:42.162 | map[id:article translation:map[date:{{ .Date }} draft:草稿 edit_title:編輯內容 reading_time:map[other:{{ .Count }} 分鐘] part:Part this_article:This Article anchor_label:定位點 date_updated:上次編輯: {{ .Date }} reading_time_title:預計閱讀時間 table_of_contents:目錄 word_count:map[one:{{ .Count }} 字 other:{{ .Count }} 字] part_of_series:This article is part of a series.]]
14:43:42.169 | Failed: build command exited with code: 255
14:43:42.971 | Failed: an internal error occurred
```

**Comment by monstermuffin** [comment]
Okay, I didn't notice CF was trying to build a prehistoric version of Hugo for some reason.

After manually specifying the latest version I'm getting this now:

```
-- | --
17:56:26.808 | Submodule 'themes/blowfish' (https://github.com/nunocoracao/blowfish.git) registered for path 'themes/blowfish'
17:56:26.809 | Cloning into '/opt/buildhome/clone/themes/blowfish'...
17:56:26.809 | Submodule path 'themes/blowfish': checked out '17557c7d731c3f07fb8cf72ebfc11c0cf21faa9a'
17:56:26.809 |  
17:56:26.857 | Success: Finished cloning repository files
17:56:27.679 | Installing dependencies
17:56:27.693 | Python version set to 2.7
17:56:31.345 | v12.18.0 is already installed.
17:56:32.594 | Now using node v12.18.0 (npm v6.14.4)
17:56:32.856 | Started restoring cached build plugins
17:56:32.876 | Finished restoring cached build plugins
17:56:33.471 | Attempting ruby version 2.7.1, read from environment
17:56:37.504 | Using ruby version 2.7.1
17:56:37.941 | Using PHP version 5.6
17:56:38.378 | 5.2 is already installed.
17:56:38.409 | Using Swift version 5.2
17:56:38.409 | Installing Hugo 0.110.0
17:56:40.689 | hugo v0.110.0-e32a493b7826d02763c3b79623952e625402b168+extended linux/amd64 BuildDate=2023-01-17T12:16:09Z VendorInfo=gohugoio
17:56:40.693 | Started restoring cached go cache
17:56:40.715 | Finished restoring cached go cache
17:56:40.878 | go version go1.14.4 linux/amd64
17:56:40.896 | go version go1.14.4 linux/amd64
17:56:40.900 | Installing missing commands
17:56:40.900 | Verify run directory
17:56:40.901 | Executing user command: hugo -b $CF_PAGES_URL
17:56:41.025 | Start building sites …
17:56:41.026 | hugo v0.110.0-e32a493b7826d02763c3b79623952e625402b168+extended linux/amd64 BuildDate=2023-01-17T12:16:09Z VendorInfo=gohugoio
17:56:42.318 | WARN 2023/01/30 17:56:42 .File.Path on zero object. Wrap it in if or with: {{ with .File }}{{ .Path }}{{ end }}
17:56:42.978 | Error: Error building site: failed to render pages: render of "home" failed: "/opt/buildhome/repo/themes/blowfish/layouts/index.html:4:7": execute of template failed: template: index.html:4:7: executing "main" at <partial $partial .>: error calling partial: "/opt/buildhome/repo/themes/blowfish/layouts/partials/home/background.html:11:97": execute of template failed: template: partials/home/background.html:11:97: executing "partials/home/background.html" at <$homepageImage.RelPermalink>: nil pointer evaluating resource.Resource.RelPermalink
17:56:42.979 | Total in 2027 ms
17:56:42.989 | Failed: build command exited with code: 255
17:56:43.841 | Failed: an internal error occurred
```

**Comment by nunocoracao** [comment]
I am sorry @MonsterMuffin this is not a bug on Blowfish. I am converting this into a discussion.

**Comment by nunocoracao** [comment]
I personally never worked with CloudFlare hope someone else can help. You can look at the workflows for Blowfish in GitHub and see if something is translatable to the CF world. At this stage, there is nothing else I can do. Are you able to run it locally?

**Reply by monstermuffin** [reply]
Thanks for looking in any case.

I think I found my issue. I deployed this incorrectly. I was making changes to the /theme directory and was not placing my images in `assets/img`. Totally my misunderstanding, I think the docs could add where exactly these images go.

I have another issue with CF pages now in that it deploys to a completely blank page. Netlify seems to deploy into a better state (still with some issues) but locally it's all good.

This can be ignored now as I don't know why CF pages deploys like this and it seems to be an issue with CF itself not blowish.

Thanks again @nunocoracao.

**Reply by nunocoracao** [reply]
@MonsterMuffin are you defining the base url correctly? I had a similar problem with firebase initially

**Comment by monstermuffin** [comment, marked as answer]
Deploy background assets into assets/img.

**Comment by davefass** [comment]
This is something Cloudflare has been working on according to [this thread](https://github.com/cloudflare/pages-build-image/discussions/1). Apparently the current build image for Pages is very out of date...hopefully they'll update soon!

In the meantime, I've had success deploying Blowfish projects via [direct upload](https://developers.cloudflare.com/pages/platform/direct-upload/), and I've seen people talking about having success using Github Actions to tie into CF Pages, but I haven't personally given that a shot yet.

**Comment by thecoducer** [comment]
Even I was facing the same issue. It seems like Cloudflare is using a prehistoric version of Hugo to run the build. I solved this issue just by explicitly specifying the Hugo version as an environment variable. 

Check this [doc](https://developers.cloudflare.com/pages/framework-guides/deploy-a-hugo-site/#use-a-specific-hugo-version).

Check which version of Hugo you're using locally and enter that as a value for the environment variable. This works!

@nunocoracao
